lambdas/file_access_patterns/lambda_for_fileDelete/index.js

```diff
@@ -20,27 +19,36 @@ exports.deleteFile = async (event) => {
   var params = {
     TableName: "V-Transfer",
     Key: {
-      PK: `USER#${reqBody.user_email}`,
-      SK: `#FILE#${reqBody.timestamp}`,
+      PK: `USER#${event.path.u_id}`,
+      SK: `FILE#${reqBody.f_timestamp}`,
     },
+    ReturnValues: 'ALL_OLD'
   };
+
   try{
-    await dynamo.delete(params).promise()
+    var fileData = await dynamo.delete(params).promise()
+  }
+  catch(err){
+    return response(err.statusCode,err.message)
+  }
 
+  try{
     await dynamo.update({
-    TableName: "V-Transfer",
-    Key: {
-      PK: `USER#${reqBody.user_email}`,
-      SK: `METADATA`, 
-    },
-    UpdateExpression: "add storage_used :size",
-    ExpressionAttributeValues: {
-      ":size": -reqBody.size,
-    },
-  }).promise()
-  return response(201,"File delete success")
-}
+      TableName: "V-Transfer",
+      Key: {
+        PK: `USER#${event.path.u_id}`,
+        SK: `METADATA`, 
+      },
+      UpdateExpression: "add storage_used :size",
+      ExpressionAttributeValues: {
+        ":size": -fileData.Attributes.size,
```

Comment:
Is the file size being decremented from user storage_used properly?

Reply:
Yes